Mint doi form

ajj11685 opened the pull request.

Commit:
create basic form

Commit:
clean up unused code and files

Commit:
form +/- is working within the wrapper

Commit:
create xml using form data

Commit:
api call to data site

ajkyffin committed:
Add requests-based MdsApi

Alberto-Alive committed:
Merge pull request #1 from ral-facilities/mds-requests-api

Add requests-based MdsApi

Commit:
set url to data site

Commit:
restyle form

Commit:
replace date field with integer field

Commit:
add placeholder to resource type field

Commit:
integer field unit test update

Commit:
redo style on form

Commit:
handle form exceptions

Commit:
add warning for unauthorised doi-sub-domains

Commit:
handle 422 datacite error

Commit:
amendments

Commit:
handle doi prefix suffix exceptions

Commit:
fix add/delete buttons bug

Commit:
style the mint url form

Commit:
add spacing for mint url form

Commit:
add orcid id

Commit:
add tests for orcid id

Commit:
fix accessibility issues

Commit:
move creator after title

Commit:
add abstract and tests

Commit:
add resource type text and tests

Commit:
add version and tests

Commit:
add stars to mandatory fields

Commit:
add page to assign url to doi

Commit:
add get post requests to add url page

Commit:
add collapsible elements to helper text

Commit:
add funder field

Commit:
add funder form

Commit:
convert dict to xml for funder

Commit:
funder form is not compulsory add tests

Commit:
add date field

Commit:
do convert to xml and add tests date field

Commit:
add helper text to the field creator

Commit:
add related identifier type

Commit:
change fields order and names

Commit:
added date formset

Commit:
change warning message

ajkyffin committed:
Fixes for python 3.6

Commit:
add specific bootstrap for the doi forms

Alberto-Alive committed:
Merge pull request #2 from ral-facilities/python3.6-fixes

Fixes for python 3.6

Commit:
add red banner: This is for testing only

Commit:
change title to h1

Commit:
left align page content

Commit:
dropdown hidden on start and specify errors from datacite

Commit:
drop down rules and label

Commit:
adapt dict-to-xml to dropdown

ajj11685 committed:
Merge branch 'mint_doi_form' into mint_doi_form

ajj11685 committed:
Merge pull request #9 from AlbertoPaulPopescu/mint_doi_form

Mint doi form

ajj11685 committed:
added validation to identifier field in form

added validation to identifier field

ajkyffin committed:
Merge pull request #12 from ral-facilities/11-metadata_overwrite_issue

metadata overwrite issue

ajj11685 committed:
finished with error messages on url,doi,identifier

ajj11685 committed:
Merge pull request #14 from ral-facilities/13-error_msgs_development

mint doi form error messages fixed

ajj11685 committed:
swapped the help toggle around so it shows by default

ajj11685 committed:
Merge pull request #16 from ral-facilities/15_invert_help_display

swapped the help toggle around so it shows by default

antony-wilson commented:
There are a number of print statements that are commented out. Please can you remove all print statments, replace them with logging if needed.

ajj11685 committed:
removed commented out print statements

antony-wilson commented:
the file `doi.db` should not have been checked in

ajj11685 committed:
removed doi.db

antony-wilson commented:
There is an issue with the inheritence in the templates. On the production system `organisation_wrapper.html` has the value `{% extends "doi_site/stfc_skin.html" %}`
In `add_url.html` and `mint_url.html` the following should be removed
```
{% block head %}
<link
  href="/static/doi_site/css/bootstrap.css"
  rel="stylesheet"
/>
{% endblock %} {% block content %}
```

`create_normal.html` will also need updating but that will not be so staight forward

ajj11685 committed:
fixed template inheritance issue so that the head information needed for the mint doi form is now moved to base.html

edited help text and removed toggle functionality

changed button colours

added more text to register/update url page

added logic to decide if to display the form depending on if the user has any sub-domains associated

plus more minor ui changes

ajj11685 committed:
moved the URL field to the same form as the metadata

ajj11685 committed:
refactored code, added text adjustments, fixed links in footer to standard ukri ones

antony-wilson committed:
add check for existing DOI back in to form

antony-wilson committed:
Merge pull request #22 from ral-facilities/21_move_url_field

21 move url field

antony-wilson committed:
format code with Black

ajj11685 committed:
cleaned up imports, renamed dict, renamed MdsApi.py

ajj11685 committed:
fix conflicts

ajj11685 committed:
remove MdsApi.py

antony-wilson committed:
update version number, 2.0.0
